### DISCUSSION TITLE
🐞 Is Github README caching broken?

### DISCUSSION THREAD

**Discussion by aleksey-hoffman** [community/community#101835] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

In short, all repo branches are showing the main branch's README.

I created v2 branch on my project with a different README file. But when I open the v2 branch page I see the README content of the main branch (sometimes it loads the proper one)

Repo main and v2 branches: 
https://github.com/aleksey-hoffman/sigma-file-manager
https://github.com/aleksey-hoffman/sigma-file-manager/tree/v2

**Comment by qu-tan-um** [comment]
Similar situation here.

**Comment by errose28** [comment]
I'm seeing the same thing on https://github.com/apache/ozone-site
Every branch has a different README but only the main branch's README is displayed. I recall this correctly showing the README specific to the current branch up until recently.

**Comment by megasanjay** [comment]
Just noticed this on my repositories. I thought I was imagining things

**Comment by gorghino** [comment]
UP. This is happening also on my repositories

**Comment by aleksey-hoffman** [comment]
Github support said they are working on it

**Reply by bassamanator** [reply]
What a relief.

Is there an issue# where this is being discussed?

**Reply by aleksey-hoffman** [reply]
Nope, got this reply from GitHub support

**Comment by naveen-98** [comment]
Hello!

GitHub uses caching mechanisms to optimize the performance of repositories, and occasionally, there might be caching issues that lead to outdated content being displayed. If you are experiencing problems with the README file not updating properly, here are some steps you can take to address the issue:

1. **Force Refresh:**
   - Perform a hard refresh on your browser to ensure that you are loading the most recent content. You can typically do this by pressing `Ctrl + F5` (Windows/Linux) or `Cmd + Shift + R` (Mac) in your browser.

2. **Check Browser Extensions:**
   - Disable any browser extensions or plugins that might interfere with the loading of GitHub content. Some extensions could cause caching issues.

3. **Clear Browser Cache:**
   - Clear your browser's cache to remove any stored data that might be causing the outdated content to be displayed.

4. **GitHub Repository Cache:**
   - GitHub itself might be caching content. Wait for some time and check if the issue resolves on its own. GitHub's caches are usually updated regularly.

5. **Branch Naming:**
   - Ensure that the branch you are working on is named correctly. Sometimes, using unconventional branch names might lead to caching issues.

6. **Explicitly Reference README:**
   - Instead of relying on GitHub's automatic README rendering, you can explicitly reference the README file in your branch, like: `https://github.com/username/repository/blob/branch/README.md`. This ensures that you are viewing the README from the specific branch.

7. **GitHub Support:**
   - If the issue persists and none of the above steps resolves it, consider reaching out to GitHub support or checking GitHub community forums for assistance. They may provide insights into any ongoing system issues or offer guidance specific to your case.

Remember that changes to repositories, especially with regards to caching, may take some time to propagate across GitHub's servers. If the issue continues, GitHub support is the best resource for resolving such technical problems.

**Reply by aleksey-hoffman** [reply]
None of these fix the problem. Github broke something internally about a week ago. It was fine before that for me and other people (see replies above)

**Comment by alinkedd** [comment]
Same/duplicate discussion is here https://github.com/orgs/community/discussions/101898

**Comment by matheeshapathirana** [comment]
Same problem for me

**Comment by bassamanator** [comment]
Same problem on my repo here: https://github.com/bassamanator/Sovol-SV06-firmware.

This repo is 1 year old. I maintain 4 branches are very similar but distinct from one another. The problem started happening about 1.5 weeks ago.

**Comment by gorghino** [comment, marked as answer]
It seems they fixed

**Reply by bassamanator** [reply]
Yep.
